python-sdk/src/astro/databases/snowflake.py

```diff
@@ -638,6 +639,22 @@ def _copy_into_table_from_stage(self, source_file, target_table, stage):
             self.drop_stage(stage)
         return rows
 
+    def _validate_before_copy_into(self, source_file, target_table, stage):
+        """Validate COPY INTO command to tests the files for errors but does not load them."""
+        if self.load_options.validation_mode is None:
+            return
```

Comment:
The if should be wrapped around the `_validate_before_copy_into` function call.

Reply:
`mypy` was complaining that this is increasing the complexity of the code. Any better way you could suggest @feluelle ?

Reply:
https://github.com/astronomer/astro-sdk/pull/1689/commits/c4a6aec82c6a4e85d0bd1605c0fc77c0a0634447